astropy/utils/console.py

```diff
@@ -218,23 +218,6 @@ def _color_text(text, color):
     return f"\033[{color_code}m{text}\033[0m"
 
 
-def _decode_preferred_encoding(s):
-    """Decode the supplied byte string using the preferred encoding
-    for the locale (`locale.getpreferredencoding`) or, if the default encoding
-    is invalid, fall back first on utf-8, then on latin-1 if the message cannot
-    be decoded with utf-8.
-    """
-    enc = locale.getpreferredencoding()
-    try:
-        try:
-            return s.decode(enc)
-        except LookupError:
-            enc = _DEFAULT_ENCODING
```

Comment:
When this function is removed `_DEFAULT_ENCODING` is used only once in `_write_with_fallback()`, so I suggest we remove the constant too.

Reply:
indeed, thanks for spotting this !